app/services/rdf_importer.rb

```diff
@@ -0,0 +1,52 @@
+class RdfImporter
+  attr_reader :url, :errors, :term_list
+  delegate :url_to_graph, :graph_to_termlist, :validators, :to => :injector
+
+  def initialize(errors)
+    @errors = errors
+  end
+
+  def call(url)
+    reset(url)
+    validators.each {|v| v.new.validate(self)}
+    build_graph
+    build_term_list
+  end
+
+  private
+
+  def reset(url)
+    errors.clear
```

Comment:
Hmm, I suppose it could be the caller's job to reset errors, but I'd still like this object not to have any contact with external errors objects.